File: Python_clients/subscriber_will.py
import paho.mqtt.client as mqtt

subscriber = mqtt.Client(client_id= "cdacsensor01" ,protocol=mqtt.MQTTv5 )

TOPIC='cdac/diot'

subscriber.username_pw_set("sensor","diot")

#set my will to the broker: TOPIC : PAYLOAD: QoS : 
WILL_TOPIC = 'device/dead'

subscriber.will_set(WILL_TOPIC,"Sensor DHT22 is offline",qos=1,retain=True)


def on_connection(Client,userdata,flags,rc):
    if rc== mqtt.MQTT_ERR_SUCCESS:
     print('connection with MQTT broker now')
     subscriber.subscribe(TOPIC)

    else:
       print("Error occured")

def on_receive(Client,userdata,msg):
   print(f'Topic : {msg.topic}, Message : {str(msg.payload)}')
  
  
subscriber.on_connect = on_connection
subscriber.on_message = on_receive


subscriber.connect("localhost",1883)

subscriber.loop_forever()

File: NodeJs Client/subscriber_will.js
/* ------------------------------------
Description : A publisher example to check will message
Broker: Mosquitto
*/

const mqtt = require('mqtt')
const TOPIC = 'cdac/diot'
const WILL_TOPIC = 'dead/device'

//connection to broker
const client = mqtt.connect('mqtt://localhost:1883',{
    protocolVersion:5,
    keepalive:0,
    clientId:'cdacdiot_sensor'
});

client.on('connect',()=>{
    console.log("i'm connected to the Broker")
    //Subscription to Broker on Topic
    client.subscribe([TOPIC,WILL_TOPIC],(err)=>{
        if(err){
            console.log("Barobar connect kr re bsdk");
            return;
        }
        console.log(`Subscriber to Topic: ${TOPIC}`)               
    })
})

//To receive messages on subscribed TOPIC
client.on('message',(topic,payload)=>{
    console.log(`Message received at TOPIC: ${topic}`)
    console.log(payload.toString())

    try {
        let data = JSON.parse(payload.toString())
        if(data && data.temperature && data.temperature>30){

            console.log("TEmperature is more than 30 Celsius")
        }
    } catch (error) {
        console.log(error)
    }
})
